src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { memo, useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  Col, Collapsible, Icon, IconButton, Form, Row,
+} from '@edx/paragon';
+import { AddComment, Delete } from '@edx/paragon/icons';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import messages from './messages';
+import { actions } from '../../../../../data/redux';
+import { answerOptionProps } from '../../../../../data/services/cms/types';
+
+let AnswerOption = ({
+  answer,
+  hasSingleAnswer,
+  // injected
+  intl,
+  // redux
+  deleteAnswer,
+  updateAnswer,
+}) => {
+  const [isFeedbackVisible, setIsFeedbackVisible] = useState(false);
+  const removeAnswer = () => deleteAnswer({ id: answer.id });
+  const setAnswer = (payload) => updateAnswer({ id: answer.id, hasSingleAnswer, ...payload });
+
+  useEffect(() => {
+    // show feedback fields if feedback is present
+    setIsFeedbackVisible(isVisible => (
+      !!answer.selectedFeedback || !!answer.unselectedFeedback || !!answer.feedback || isVisible
+    ));
+  }, [answer]);
+
+  const toggleFeedback = (open) => {
+    // do not allow to hide if feedback is added
+    if (!!answer.selectedFeedback || !!answer.unselectedFeedback || !!answer.feedback) {
+      setIsFeedbackVisible(true);
+      return;
+    }
+    setIsFeedbackVisible(open);
+  };
+
+  const Checker = () => {
+    let CheckerType = Form.Checkbox;
+    if (hasSingleAnswer) {
+      CheckerType = Form.Radio;
+    }
+    return (
+      <CheckerType
+        className="pl-4 mt-3"
+        value={answer.id}
+        onChange={(e) => setAnswer({ correct: e.target.checked })}
+        defaultChecked={answer.correct}
+      >
+        {answer.id}
+      </CheckerType>
+    );
+  };
+
+  const feedbackControl = ({
+    feedback, onChange, labelMessage, labelMessageBoldUnderline,
+  }) => (
+    <Form.Group>
+      <Form.Label className="mb-3">
+        <FormattedMessage
+          {...labelMessage}
+          values={{
+            answerId: answer.id,
+            boldunderline: <b><u><FormattedMessage {...labelMessageBoldUnderline} /></u></b>,
+          }}
+        />
+      </Form.Label>
+      <Form.Control
+        placeholder={intl.formatMessage(messages.feedbackPlaceholder)}
+        value={feedback}
+        onChange={onChange}
+      />
+    </Form.Group>
+  );
+
+  const displayFeedbackControl = () => {
+    if (typeof answer.feedback !== 'undefined') {
```

Comment:
We can use `.?` and avoid typeof what do you say?

Reply:
@farhaanbukhsh Used `lodash.isUndefined` to check here. Also implement all other suggestions.